dissect/target/tools/info.py

```diff
@@ -90,49 +90,51 @@ def main():
 
 
 def get_target_info(target: Target) -> dict[str, Union[str, list[str]]]:
-    return {
+    info = {
+        "disks": get_disks_info(target),
+        "volumes": get_volumes_info(target),
+        "children": get_children_info(target),
         "hostname": target.hostname,
-        "domain": target.domain,
+        "domain": None,
         "ips": target.ips,
         "os_family": target.os,
         "os_version": target.version,
         "architecture": target.architecture,
-        "language": target.language,
-        "timezone": target.timezone,
-        "install_date": target.install_date,
-        "last_activity": target.activity,
-        "disks": get_disks_info(target),
-        "volumes": get_volumes_info(target),
-        "children": get_children_info(target),
+        "language": None,
+        "timezone": None,
+        "install_date": None,
+        "activity": None,
     }
 
+    optional = ["domain", "language", "timezone", "install_date", "activity"]
+
+    for func in optional:
+        if target.has_function(func):
+            info[func] = getattr(target, func)
```

Comment:
```suggestion
        info[func] = getattr(target, func, None)
```

Would this work?
In this form you could also assign it immediately in the dict without the need to loop over a list of optional keys.

Reply:
Unfortunately this would not work as the `getattr` would fail when calling an incompatible or non-existing plugin for the current target. Which is why the `has_function` method exists.

Reply:
I've added a small improvement regarding your suggestion in [1cc7663](https://github.com/fox-it/dissect.target/pull/405/commits/1cc7663a82c8a520771a0357eaa3c4ba745a9fba).